src/main/java/pingis/controllers/UserController.java

```diff
@@ -52,10 +56,12 @@ public String user(Model model, Principal principal) {
     model.addAttribute("user", user);
 
     MultiValueMap<Challenge, TaskInstance> myTasksInChallenges
-            = challengeService.getCompletedTaskInstancesInUnfinishedChallenges();
-    TaskInstance lastUnfinished = taskInstanceService.getLastUnfinishedInstance();
-    List<TaskInstance> history = taskInstanceService.getHistory();
-    List<Challenge> availableChallenges = challengeService.getAvailableChallenges(
+            = userService
```

Comment:
Implement the dashboardService.getAvailableChallenges(userService.getCompletedTaskInstancesInUnfinishedChallenges()) as it's own method in DashboardService.

Reply:
Delegate this to Ville. He's doing dashboard-related stuff anyway.